### PR TITLE
feat(server): connect thread with asset

### PR DESCRIPTION
**Pull request by nourbalaha**
# Overview
I connected thread with asset

## What I did
- Now When Uploading a new asset, a new thread is being created and saved into the database and then connected to asset via `asset.threadId`
- I made `asset.threadId` mandatory when creating a new asset via asset builder.

### CODE REVIEW

File: server/pkg/asset/builder_test.go [hunk 1]
```diff
@@ -122,6 +123,21 @@ func TestBuilder_Build(t *testing.T) {
...
+				project:     pid,
+				createdBy:   uid,
+				fileName:    "hoge",
+				size:        size,
+				previewType: PreviewTypeFromRef(lo.ToPtr(PreviewTypeImage.String())),
+				file:        &f,
+				uuid:        "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
+				thread:      ThreadID{},
+			},
+			err: ErrZeroSize,
```
Comment (rot1024) [inline server/pkg/asset/builder_test.go:139]: This test name and `err: ErrZeroSize` is mismatched. Does it test `ErrNoThread`?

File: server/internal/usecase/interactor/asset.go [hunk 2]
```diff
@@ -78,6 +79,14 @@ func (i *Asset) Create(ctx context.Context, inp interfaces.CreateAssetParam, ope
 				return nil, err
 			}
 
+			th, err := thread.New().NewID().Workspace(prj.Workspace()).Comments([]*thread.Comment{}).Build()
```
Comment (rot1024) [inline server/internal/usecase/interactor/asset.go:82]:
```suggestion
			th, err := thread.New().NewID().Workspace(prj.Workspace()).Build()
```

Note that a nil slice and an empty slice is almost same in Golang

**Review by rot1024 — APPROVED** [review]

File: server/pkg/asset/builder_test.go (hunk 1 above)
Comment (nourbalaha) [reply server/pkg/asset/builder_test.go]: This is done by mistake, I pushed a fix. Thanks!

File: server/internal/usecase/interactor/asset.go (hunk 2 above)
Comment (nourbalaha) [reply server/internal/usecase/interactor/asset.go]: Done!

**Pull request merged**